src/harness/simple_crl_server.py

```diff
@@ -0,0 +1,331 @@
+                                         cert_name)
+  command_exit_status = subprocess.call(revoke_cert_command, shell=True)
+  if command_exit_status:
+    raise Exception("RunTimeError:Certificate Revocation is failed:"
+                    "exit_code:{}".format(command_exit_status))
+  logging.info('%s is blacklisted successfully', cert_name)
+
+
+def updateCrlUrlAndRegenerateCertificates():
+  """Updates the CDP  field and re-generates certificates.
+
+  Updates URI.0 field in the in openssl.cnf file with the CRL URL of this simple CRL
+  server. Invokes the generate_fake_certs.sh script to regenerate all certificates.
+  """
+
+  # Update CRL URL in openssl.cnf.
+  crl_url = 'URI.0 = {}'.format(DEFAULT_CRL_URL)
+  update_crl_section_command = (r"sed -i -e 's~^URI.0\s=\s.*$~%s~1' ../../cert/openssl.cnf"
+                                % crl_url)
+  command_exit_status = subprocess.call(update_crl_section_command, shell=True)
+  if command_exit_status:
+    raise Exception("RunTimeError:Could not update CRL URL in openssl.cnf:"
+                    "exit_code:{}".format(command_exit_status))
+  logging.info('CRL URL has been updated correctly in openssl.cnf')
+
+  # Re-generate certificates.
+  script_name = './generate_fake_certs.sh'
+  command = 'cd {0} && {1}'.format(getCertsDirectoryAbsolutePath(), script_name)
+  command_exit_status = subprocess.call(command, shell=True)
+  if command_exit_status:
+    raise Exception("RunTimeError:Regeneration of certificates failed:"
+                    "exit_code:{}".format(command_exit_status))
+  logging.info("The certs are regenerated successfully")
+
+  # Generate CRL Chain.
+  generateCrlChain()
+
+
+def generateCrlChain():
+  """ Generate CRL Chain for the intermediate CAs.
+
+  It creates a CRL file for the intermediate CAs (CBSD CA, DP CA and SAS CA) and the root CA.
+  All the CRL files are concatenated to create CRL chain file named ca.crl and
+  placed in the 'harness/certs/crl/' directory.
+  """
+
+  create_crl_chain_command = "cd {0} && ./revoke_and_generate_crl.sh -u".format(
+      getCertsDirectoryAbsolutePath())
+  command_exit_status = subprocess.call(create_crl_chain_command, shell=True)
+  if command_exit_status:
+    raise Exception("RunTimeError:Creation CRL chain is failed:"
+                    "exit_code:{}".format(command_exit_status))
+  if not os.path.exists(os.path.join(getCertsDirectoryAbsolutePath(),
+                                     'crl/ca.crl')):
+    raise Exception("RunTimeError:crl/ca.crl is not found:"
+                    "exit_code:{}".format(command_exit_status))
+  logging.info('crl/ca.crl CRL chain is created successfully')
+
+
+def crlServerStart():
+  """Start CRL server."""
+
+  # Create CRL chain containing CRLs of sub CAs and root CA.
+  generateCrlChain()
+
+  try:
+    crl_server = SimpleCrlServer(DEFAULT_CRL_URL, os.path.join(getCertsDirectoryAbsolutePath(),
+                                                             'crl/ca.crl'))
```

Comment:
Indent +2 spaces, to preserve alignment.

Reply:
fixed. thanks.